### ISSUE TITLE
Internal compiler error message during Flutter hot reload

### ISSUE THREAD

**Issue by timsneath** [dart-lang/sdk#35980]
Internal compiler error message during Flutter hot reload:
```
Compiler message:
org-dartlang-debug:synthetic_debug_expression:1:1: Error: A prefix can't be used as an expression.
http
^^^^
org-dartlang-debug:synthetic_debug_expression:1:1: Error: Getter not found: 'using'.
using
^^^^^
org-dartlang-debug:synthetic_debug_expression:1:1: Error: The getter 'using' isn't defined for the class '_MyHomePageState'.
 - '_MyHomePageState' is from 'package:bookmarker/main.dart' ('lib/main.dart').
Try correcting the name to the name of an existing getter, or defining a getter or field named 'using'.
using
```

`dart --version` shows:
```
Dart VM version: 2.1.2-dev.0.0.flutter-c92d5ca288 (Wed Feb 13 19:14:26 2019 +0000) on "macos_x64"
```

Repo at the time of the error is here:
https://github.com/timsneath/bookmarker/commit/5b7348e7803242a70f964418e2d9d4d80228f1db

(Let me know if you need access -- it's private, but happy to add Googlers.)

**Comment by aam** [comment]
How easy it is to reproduce?
Does it happen when using IJ? Hot reloading while sitting on a breakpoint? Changing particular file?

**Comment by timsneath** [comment]
Sadly, it was a one-off. Haven't seen it repeatedly. I wonder if there's more info we can capture in the compiler message to be able to track this kind of thing more effectively (can we dump a log, for instance?)

**Comment by aam** [comment]
The fact that it references `synthetic_debug_expression` points to problem with expression evaluation - perhaps to display content of variables when you pause on breakpoint. Compilation errors themselves and the fact that they reference `http` and `using` words point to possible dart vm observatory communication breakdown between device and ide.

If I'm not mistaken IJ Flutter plugin has `Enable verbose logging` option that will make verbose content of plugin<->vm interface.

cc @devoncarew for more ideas

**Comment by DanTup** [comment]
If using VS Code, there are a number of [log files](https://dartcode.org/docs/logging/#individual-logs) you can enable permanently. They get overwritten on each new session (for the debugger one, that means on each debug session, not VS Code session) so they won't grow indefinitely.

I have them enabled always, so if I see an issue like this, I can grab a copy of them without needing to be able to repro. In this instance, the Flutter Run and Observatory logs might have been useful - it would've included information about breakpoints, watched variables, etc.

**Comment by iampawan** [comment]
@timsneath Happens with me too, but "flutter clean" solves this for me.

**Comment by DanTup** [comment]
I think this is the same as https://github.com/dart-lang/sdk/issues/36091 that is fixed in Dart and rolled into Flutter. If I understand correctly, it's re-reporting errors from previous evaluations (including those from the watch window, etc.).

Is anyone still seeing it on `master` since 13 days ago? https://github.com/flutter/flutter/commit/a29906d5f552f2984daf9bef569b7fe1295e1c70

**Comment by a-siva** [comment]
This issue is fixed at  flutter/flutter@a29906d
